src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java

```diff
@@ -448,6 +453,39 @@ public static void getBackpack(ItemStack item, Consumer<PlayerBackpack> callback
         }
     }
 
+    public boolean isProtected(ProtectionType type) {
+        int armorCount = 0;
+        boolean first = true;
+
+        NamespacedKey setId = null;
+        for (HashedArmorpiece armor : armor) {
+            Optional<SlimefunArmorPiece> armorPiece = armor.getItem();
+            if (!armorPiece.isPresent()) return false;
+
+            if (armorPiece.get() instanceof CustomProtection) {
+                CustomProtection protectedArmor = (CustomProtection) armorPiece.get();
+
+                if (first) {
+                    if (protectedArmor.isFullSetRequired()) setId = armorPiece.get().getSetId();
+                    first = false;
```

Comment:
Instead of a first variable you could literally just check if the ID is null and set it.